backend/scripts/transfer-contract-ownership.ts

```diff
@@ -1,37 +1,117 @@
+    alias: 's',
+    type: 'array',
+    description: 'List of slugs to transfer',
+    demandOption: true,
+  })
+  .option('toUserId', {
+    alias: 'u',
+    type: 'string',
+    description: 'User ID to transfer contracts to',
+    demandOption: true,
+  })
+  .parseSync() as Args
+
+const BATCH_SIZE = 20
 
 if (require.main === module) {
-  runScript(async ({ pg, firestore }) => {
-    const fromUserId = 'zgCIqq8AmRUYVu6AdQ9vVEJN8On1'
-    const contractIdsToTransfer = await pg.map(
-      `select id from contracts where creator_id = $1`,
-      [fromUserId],
-      (row) => row.id
-    )
-    const toUserId = '4juQfJkFnwX9nws3dFOpz4gc1mi2'
-    const toUser = await pg.one(
-      `select * from users where id = $1`,
-      [toUserId],
-      (row) => convertUser(row)
-    )
-    console.log(
-      `Transferring ${contractIdsToTransfer.length} contracts from ${fromUserId} to ${toUser.name}`
-    )
-    const writer = new SafeBulkWriter()
-    for (const id of contractIdsToTransfer) {
-      const doc = firestore.collection('contracts').doc(id)
-      writer.update(doc, {
-        creatorId: toUser.id,
-        creatorAvatarUrl: toUser.avatarUrl,
-        creatorUsername: toUser.username,
-        creatorName: toUser.name,
-        creatorCreatedTime: toUser.createdTime,
-      })
-    }
-    await writer.close()
+  runScript(async () => {
+    const { slugs: slugsToTransfer, toUserId } = argv
+
+    try {
+      const toUserData = await supabaseDirectClient.oneOrNone<User>(
+        'SELECT id, username, name, created_time FROM users WHERE id = $1',
+        [toUserId]
+      )
 
-    console.log('done.')
+      if (!toUserData) {
+        console.error('User not found')
+        return
+      }
+
+      const toUser: User = toUserData
+
+      const contractsData = await supabaseDirectClient.any<Contract>(
+        'SELECT id, data FROM contracts WHERE slug IN ($1:csv)',
+        [slugsToTransfer]
+      )
+
+      if (contractsData.length === 0) {
+        console.error('No contracts found')
+        return
+      }
+
+      const contractsToTransfer: Contract[] = contractsData
+
+      console.log(
+        `Transferring ${contractsToTransfer.length} contracts to ${toUser.name}`
+      )
+
+      for (let i = 0; i < contractsToTransfer.length; i += BATCH_SIZE) {
+        const batch = contractsToTransfer.slice(i, i + BATCH_SIZE)
+
+        const updates = batch.map(({ id, data }) => {
+          const description = data.description || ''
+          const creatorUsername = data.creatorUsername || 'Unknown creator'
+          const creatorName = data.creatorName || 'Unknown name'
+
+          const updatedDescription = `${
+            description ? description : ''
+          }\n\nThis market was created by ${creatorUsername} (${creatorName}) and has changed ownership.`
+
+          return {
+            id,
+            creator_id: toUser.id,
+            created_time: toUser.created_time,
+            data: {
+              ...data,
```

Comment:
you should only update the fields you want to update, and not rewrite the entire data column. Someone could bet in between when you read and write and now you've overwrriten their change to the contract with old data.